clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -484,6 +502,12 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
 #endif
         shard->prev_instr_ = memref;
+        if (next_pc != nullptr) {
+            shard->prev_instr_was_syscall_ = instr_is_syscall(&cur_instr_decoded);
+            shard->prev_instr_decoded_ = &cur_instr_decoded;
```

Comment:
As discussed offline: `cur_instr_decoded` needs to be heap-allocated. `&cur_instr_decoded` will go out of scope when this iteration ends, which means `prev_instr_decoded_` will become a dangling pointer.